FirebaseMLModelDownloader/Tests/Integration/ModelDownloaderIntegrationTests.swift

```diff
@@ -176,4 +182,167 @@ final class ModelDownloaderIntegrationTests: XCTestCase {
     waitForExpectations(timeout: 5, handler: nil)
     XCTAssertEqual(modelDownloadManager.downloadStatus, .completed)
   }
+
+  func testGetModel() {
+    let testApp = FirebaseApp.app()!
+    let testName = "model-downloader-test"
+    let testModelName = "image-classification"
+
+    let conditions = ModelDownloadConditions()
+    let modelDownloader = ModelDownloader.modelDownloader(
+      app: testApp,
+      defaults: .getTestInstance(testName: testName)
+    )
+
+    /// Test download type - latest model.
+    var downloadType: ModelDownloadType = .latestModel
+    let latestModelExpectation = expectation(description: "Get latest model.")
+
+    modelDownloader.getModel(
+      name: testModelName,
+      downloadType: downloadType,
+      conditions: conditions,
+      progressHandler: { progress in
+        XCTAssertNotNil(progress)
```

Comment:
Hmm is progress optional? If so, should it be?

Reply:
So the handler itself can be optional. But if it exists, it's called within urlSession didWriteData with a non-optional progress.

Reply:
Gotcha, so this could never fail. Maybe a more appropriate check here would be to ensure the value is within the expected range (0-1)?

Reply:
Oh sorry I understand now, so progress can be null because the handler isn't called at all (which happens if the model is already on device). So this is testing that the model is indeed being downloaded. I like the range idea too.